web-common/src/features/chat/core/chat.ts

```diff
@@ -0,0 +1,596 @@
+      }
+    },
+    null,
+  );
+
+  // Instance-specific actions
+  const coreActions = {
+    createNewConversation(): void {
+      currentConversationId.set(null);
+    },
+
+    selectConversation(conv: V1Conversation): void {
+      currentConversationId.set(conv.id || null);
+    },
+
+    async sendMessage(messageText: string): Promise<string | null> {
+      if (!messageText.trim()) return null;
+
+      const $runtime = get(runtime);
+      const currentId = get(currentConversationId);
+
+      // Detect context based on current page
+      const appContext = detectAppContext(get(page));
+
+      try {
+        if (!currentId) {
+          // No current conversation - create a new one
+          const $completeNewConversationMutation = get(completeNewConversation);
+          const response = await $completeNewConversationMutation.mutateAsync({
+            instanceId: $runtime.instanceId,
+            data: {
+              // No conversationId for new conversations
+              appContext: appContext || undefined,
+              messages: [{ role: "user", content: [{ text: messageText }] }],
+            },
+          });
+          return response.conversationId || null;
+        } else {
+          // Existing conversation - add message to it
+          const $completeExistingConversationMutation = get(
+            completeExistingConversation,
+          );
+          await $completeExistingConversationMutation.mutateAsync({
+            instanceId: $runtime.instanceId,
+            data: {
+              conversationId: currentId,
+              messages: [{ role: "user", content: [{ text: messageText }] }],
+            },
+          });
+          return currentId;
+        }
+      } catch (e) {
+        console.error("Failed to send message:", e);
+        throw e;
+      }
+    },
+  };
+
+  // Queries
+  const listConversationsQueryOptionsStore = derived([runtime], ([$runtime]) =>
+    getRuntimeServiceListConversationsQueryOptions($runtime.instanceId),
+  );
+  const listConversationsQuery = createQuery(
+    listConversationsQueryOptionsStore,
+  );
+
+  const getConversationQueryOptionsStore = derived(
+    [runtime, currentConversationId],
+    ([$runtime, $currentConversationId]) =>
+      getRuntimeServiceGetConversationQueryOptions(
+        $runtime.instanceId,
+        $currentConversationId || "",
+        undefined,
+        {
+          query: {
+            enabled:
+              !!$currentConversationId &&
+              !isOptimisticConversationId($currentConversationId),
+            staleTime: 5 * 60 * 1000, // 5 minutes
+            gcTime: 10 * 60 * 1000, // 10 minutes
+          },
+        },
+      ),
+  );
+  const getConversationQuery = createQuery(getConversationQueryOptionsStore);
+
+  // Derived state
+  const listConversationsData = derived(
+    listConversationsQuery,
+    ($query) => $query.data,
+  );
+  const getConversationData = derived(
```

Comment:
`getConversationData` is not used anywhere. Feels like over kill to have this explanatory variable. Would not be hard understand if `currentConversation` has `$query.data?.conversation || null`.